src/NzbDrone.Common/UniqueIdentifier/UniqueIdentifierGenerator.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using NzbDrone.Common.Cache;
+
+namespace NzbDrone.Common.UniqueIdentifier
+{
+    public interface IUniqueIdentifierGenerator
+    {
+        int Get(string key);
+    }
+
+    public class UniqueIdentifierGenerator : IUniqueIdentifierGenerator
+    {
+        private static int _lastIdentifier;
+
+        private ICached<int> _cache;
+
+        public UniqueIdentifierGenerator(ICacheManager cacheManager)
+        {
+            _cache = cacheManager.GetCache<int>(typeof(UniqueIdentifierGenerator));
+        }
+
+        public int Get(string key)
+        {
+            return _cache.Get(key, GetNextIdentifier, TimeSpan.FromHours(24));
```

Comment:
if you do abs it still gives you  2,147,483,647 unique values without all this complication (no need to cache and they even stay the same across restarts)